prototype/clustering/clustering.ino

```diff
@@ -0,0 +1,65 @@
+#define REAL_NUMBER float
+#define INTEGER uint16_t
+#define DIGITS_PRECISION 3
+
+#include "stats.h"
+#include "arduino_serial_port.hpp"
+
+enum State {
+  READING,
+  PROCESSING,
+};
+
+State state = State::READING;
+
+void setup() {
+
+    Serial.begin(9600);
+}
+
+uint16_t nbPoints = 0;
+uint16_t nbClusters = 0;
+uint16_t dimensions = 0;
+REAL_NUMBER* data;
+
+void read() {
+
+  if (Serial.available()) {
+    readElement(&nbPoints, sizeof(nbPoints));
+    readElement(&nbClusters, sizeof(nbClusters));
+    readElement(&dimensions, sizeof(dimensions));
+    data = (REAL_NUMBER*) malloc(sizeof(REAL_NUMBER) * nbPoints * dimensions);
```

Comment:
Why do you cast using the define and not the typedef real_numbers?

Reply:
True, I will change it